src/service/valid.ts

```diff
@@ -0,0 +1,201 @@
+import { propsInput, regExp, errorMessage } from './propsInput';
+
+interface IValidMessage {
+    message: string;
+    regExp?: RegExp;
+    isValid: (value: any) => boolean;
+}
+
+interface InputIsNotValid {
+    input: IValidMessage;
+    target: HTMLInputElement;
+    value: any;
+    message: string;
+}
+
+interface IBlurProps {
+    target: HTMLInputElement;
+    value: any;
+    name: string;
+}
+
+interface ITarget {
+    target: HTMLInputElement;
+}
+
+export const getResForm = (form: string) => {
+    const dataObject: Record<string, any> = {};
+    const formElement = document.querySelector(form) as HTMLFormElement;
```

Comment:
Cтарайся по возможности избегать явного приведения типов, потому что это "обход" подсказок typescript'a. К тому же querySelector имеет прекрасный дженерик
```suggestion
    const formElement = document.querySelector<HTMLFormElement>(form);
```